File: product-web-app/service/bandValidator.ts
import Band from '../model/band.js';

export default class BandValidator {
  validateBand(band: Band) {
    if (band.name.length > 64) {
      return 'Name longer than 64 characters';
    }
    if (band.name.length < 2) {
      return 'Name shorter than 2 characters';
    }
    return null;
  }

  validateBandLevel(band: Band) {
    if (band.level >= 0 && band.level <= 9) {
      return 'Correct Level input';
    }
    return null;
  }
}

File: product-backend/src/main/java/org/kainos/ea/service/BandValidator.java
package org.kainos.ea.service;

import org.kainos.ea.model.BandRequest;

import java.util.Optional;

public class BandValidator {

    public Optional<String> isValidBandName(BandRequest band) {

        if (band.getName().length() > 64) {
            return Optional.of("Name longer than 64 characters");
        } else if (band.getName().length() < 2) {
            return Optional.of("Name shorter than 2 characters");
        }
        return Optional.empty();
    }

    public Optional<String> isValidBandLevel(BandRequest band) {

        if (band.getLevel() >= 0 && band.getLevel() <= 9) {
            return Optional.empty();
        } else {
            return Optional.of("Incorrect Level input");
        }
    }
}
